File: examples/main.example.py
# Function to convert a binary string to a decimal integer
def binary_to_decimal(binary):
    return int(binary, 2)  # Convert binary string to integer using base 2

# Function to convert a decimal integer to a binary string
def decimal_to_binary(decimal):
    return bin(decimal)[2:]  # Convert integer to binary string, remove the '0b' prefix

# Function to perform binary addition
def binary_addition(bin1, bin2):
    num1 = binary_to_decimal(bin1)  # Convert binary to decimal
    num2 = binary_to_decimal(bin2)
    sum_result = num1 + num2  # Perform addition
    return decimal_to_binary(sum_result)  # Convert the result back to binary

# Function to perform binary subtraction
def binary_subtraction(bin1, bin2):
    num1 = binary_to_decimal(bin1)  # Convert binary to decimal
    num2 = binary_to_decimal(bin2)
    difference = num1 - num2  # Perform subtraction
    return decimal_to_binary(difference)  # Convert the result back to binary

# Function to perform binary multiplication
def binary_multiplication(bin1, bin2):
    num1 = binary_to_decimal(bin1)  # Convert binary to decimal
    num2 = binary_to_decimal(bin2)
    product = num1 * num2  # Perform multiplication
    return decimal_to_binary(product)  # Convert the result back to binary

# Example usage
bin1 = "1010"  # Binary for 10
bin2 = "110"   # Binary for 6

# Perform operations and display results
print("Addition:", binary_addition(bin1, bin2))         # Output: "10000"
print("Subtraction:", binary_subtraction(bin1, bin2))   # Output: "100"
print("Multiplication:", binary_multiplication(bin1, bin2))  # Output: "111100"

File: examples/main.example.cs
using System;

class BinaryOperations
{
    // Function to convert binary string to decimal integer
    static int BinaryToDecimal(string binary)
    {
        return Convert.ToInt32(binary, 2); // Convert binary string to integer
    }

    // Function to convert decimal integer to binary string
    static string DecimalToBinary(int decimalNumber)
    {
        return Convert.ToString(decimalNumber, 2); // Convert integer to binary string
    }

    // Function to perform binary addition
    static string BinaryAddition(string bin1, string bin2)
    {
        int num1 = BinaryToDecimal(bin1); // Convert binary to decimal
        int num2 = BinaryToDecimal(bin2);
        int sum = num1 + num2; // Perform addition
        return DecimalToBinary(sum); // Convert result back to binary
    }

    // Function to perform binary subtraction
    static string BinarySubtraction(string bin1, string bin2)
    {
        int num1 = BinaryToDecimal(bin1); // Convert binary to decimal
        int num2 = BinaryToDecimal(bin2);
        int difference = num1 - num2; // Perform subtraction
        return DecimalToBinary(difference); // Convert result back to binary
    }

    // Function to perform binary multiplication
    static string BinaryMultiplication(string bin1, string bin2)
    {
        int num1 = BinaryToDecimal(bin1); // Convert binary to decimal
        int num2 = BinaryToDecimal(bin2);
        int product = num1 * num2; // Perform multiplication
        return DecimalToBinary(product); // Convert result back to binary
    }

    static void Main()
    {
        string bin1 = "1010"; // Binary for 10
        string bin2 = "110";  // Binary for 6

        // Perform operations and display results
        Console.WriteLine("Addition: " + BinaryAddition(bin1, bin2));         // Output: "10000"
        Console.WriteLine("Subtraction: " + BinarySubtraction(bin1, bin2));   // Output: "100"
        Console.WriteLine("Multiplication: " + BinaryMultiplication(bin1, bin2)); // Output: "111100"
    }
}
